app/logic/Mail/IMAP/IMAPFolder.ts

```diff
@@ -645,4 +646,63 @@ export class IMAPFolder extends Folder {
   newEMail(): IMAPEMail {
     return new IMAPEMail(this);
   }
+
+  async getPermissions(): Promise<ArrayColl<IMAPPermission> | undefined> {
+    if (!await this.account.hasCapability("ACL")) {
+      return undefined;
+    }
+    let conn = await this.account.connection();
+    let attributes: Array<{ type: string, value: string }>;
+    let permissions = new ArrayColl<IMAPPermission>();
+    await conn.exec('GETACL', [{ type: 'ATOM', value: this.path }], { untagged: { async ACL(untagged) { attributes = untagged.attributes; } } });
+    for (let i = 0; i < attributes.length; i += 2) {
+      let name = attributes[i].value;
+      let rights = attributes[i + 1].value;
+      let emailAddress = name.includes('@') ? name : name + this.account.emailAddress.slice(this.account.emailAddress.indexOf('@'));
+      permissions.add(new IMAPPermission(emailAddress, name, rights));
+    }
+    return permissions;
+  }
+
+  async setPermission(permission: IMAPPermission, right: keyof typeof IMAPACL) {
+    let conn = await this.account.connection();
+    await conn.exec('SETACL', [{ type: 'ATOM', value: this.path }, { type: 'ATOM', value: permission.name }, { type: 'ATOM', value: permission[IMAPACL[right]] ? "+" + right : "-" + right }]);
+  }
+
+  async removePermission(permission: PersonUID) {
+    let conn = await this.account.connection();
+    await conn.exec('DELETEACL', [{ type: 'ATOM', value: this.path }, { type: 'ATOM', value: permission.name }]);
+  }
+}
+
+export enum IMAPACL {
+  l = "lookup",
+  r = "read",
+  s = "seen",
+  w = "write",
+  i = "create",
+  k = "subfolder",
+  x = "remove",
+  t = "delete",
+  e = "expunge",
+  a = "owner",
+}
+
+export class IMAPPermission extends PersonUID {
```

Comment:
An `IMAPPermission` should *have* a person, not *be* a person.

```suggestion
export class IMAPPermission {
  for: PersonUID;
```

Reply:
Well, we'll see what what your final UI does, but for now, I'm using PersonsAutocomplete to hold the permitted users, and that needs the permissions to derive from, not aggregate, PersonUID.